google/resource_compute_instance.go

```diff
@@ -1581,6 +1678,24 @@ func suppressEmptyGuestAcceleratorDiff(d *schema.ResourceDiff, meta interface{})
 	return nil
 }
 
+func desiredStatusDiff(diff *schema.ResourceDiff, meta interface{}) error {
+	// when creating an instance, name is not set
+	oldName, _ := diff.GetChange("name")
```

Comment:
I don't know if this is the best way to check if we are creating an instance, but this is a solution inspired by what is done in `google_bigtable_instance.go` (https://github.com/terraform-providers/terraform-provider-google/blob/07a8c1f9f7428cc490ed92071484797ab7dfe828/google/resource_bigtable_instance.go#L341).

Reply:
Yep, this is the right way to check!